File: Módulo Python/Códigos/5 - Múltipla Escolha/Exercicio.02.py
#2. Para eleição de representantes de classe de uma universidade há três candidatos. Os votos são informados através de código:
# 1,2 ou 3, voto para os respectivos candidatos, 5 voto nulo e 6 voto em branco. Faça um algoritmo que calcule e escreva:
# a. A porcentagem e o total de votos para cada candidato
# b. Total de votos nulos
# c. Total de votos em branco
# d. Percentual de votos em brancos e nulos
# e. Classificação dos candidatos
# f. Total de votos

votos_candidato_1 = 0
votos_candidato_2 = 0
votos_candidato_3 = 0
votos_nulos = 0
votos_em_branco = 0
total_votos = 0

while True:
    voto = int(input("Digite o código do candidato (1, 2, 3), 5 para voto nulo, 6 para voto em branco, ou -1 para encerrar: "))

    if voto == -1:
        break

    if voto == 1:
        votos_candidato_1 += 1
    elif voto == 2:
        votos_candidato_2 += 1
    elif voto == 3:
        votos_candidato_3 += 1
    elif voto == 5:
        votos_nulos += 1
    elif voto == 6:
        votos_em_branco += 1

    total_votos += 1

porcentagem_candidato_1 = (votos_candidato_1 / total_votos) * 100
porcentagem_candidato_2 = (votos_candidato_2 / total_votos) * 100
porcentagem_candidato_3 = (votos_candidato_3 / total_votos) * 100
porcentagem_nulos = (votos_nulos / total_votos) * 100
porcentagem_em_branco = (votos_em_branco / total_votos) * 100

print("Resultado da eleição:")
print("Total de votos para o candidato 1:", votos_candidato_1)
print("Total de votos para o candidato 2:", votos_candidato_2)
print("Total de votos para o candidato 3:", votos_candidato_3)
print("Total de votos nulos:", votos_nulos)
print("Total de votos em branco:", votos_em_branco)
print("Percentual de votos nulos:", porcentagem_nulos, "%")
print("Percentual de votos em branco:", porcentagem_em_branco, "%")

if votos_candidato_1 > votos_candidato_2 and votos_candidato_1 > votos_candidato_3:
    print("O candidato 1 foi o mais votado.")
elif votos_candidato_2 > votos_candidato_1 and votos_candidato_2 > votos_candidato_3:
    print("O candidato 2 foi o mais votado.")
elif votos_candidato_3 > votos_candidato_1 and votos_candidato_3 > votos_candidato_2:
    print("O candidato 3 foi o mais votado.")
else:
    print("Houve um empate entre os candidatos.")

print("Total de votos:", total_votos)

File: Livro - Módulo Java/5 - Múltipla Escolha/Exercicio.02.java
// 2 - Para eleição de representantes de classe de uma universidade há três candidatos.
// Os votos são informados através de código: 1,2 ou 3 (voto para os respectivos candidatos),
// 5 (voto nulo) e 6 (voto em branco). Faça um algoritmo que calcule e escreva: 
// a. A porcentagem e o total de votos para cada candidato 
// b. Total de votos nulos 
// c. Total de votos em branco 
// d. Percentual de votos em brancos e nulos 
// e. Classificação dos candidatos 
// f. Total de votos 

import java.util.Scanner;

public class EleicaoRepresentantes {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int votosCandidato1 = 0;
        int votosCandidato2 = 0;
        int votosCandidato3 = 0;
        int votosNulos = 0;
        int votosBranco = 0;
        int totalVotos = 0;

        while (true) {
            System.out.println("Digite o código do voto (1, 2, 3, 5 ou 6) ou 0 para encerrar:");
            int codigoVoto = scanner.nextInt();

            if (codigoVoto == 0) {
                break;
            }

            switch (codigoVoto) {
                case 1:
                    votosCandidato1++;
                    break;
                case 2:
                    votosCandidato2++;
                    break;
                case 3:
                    votosCandidato3++;
                    break;
                case 5:
                    votosNulos++;
                    break;
                case 6:
                    votosBranco++;
                    break;
                default:
                    System.out.println("Código de voto inválido. Tente novamente.");
                    continue; 
            }

            totalVotos++;
        }

        double porcentagemCandidato1 = (double) votosCandidato1 / totalVotos * 100;
        double porcentagemCandidato2 = (double) votosCandidato2 / totalVotos * 100;
        double porcentagemCandidato3 = (double) votosCandidato3 / totalVotos * 100;
        double porcentagemNulos = (double) votosNulos / totalVotos * 100;
        double porcentagemBranco = (double) votosBranco / totalVotos * 100;

        System.out.println("a. Porcentagem de votos para cada candidato:");
        System.out.println("   Candidato 1: " + porcentagemCandidato1 + "%");
        System.out.println("   Candidato 2: " + porcentagemCandidato2 + "%");
        System.out.println("   Candidato 3: " + porcentagemCandidato3 + "%");

        System.out.println("b. Total de votos nulos: " + votosNulos);
        System.out.println("c. Total de votos em branco: " + votosBranco);

        System.out.println("d. Percentual de votos nulos e em branco: " + (porcentagemNulos + porcentagemBranco) + "%");

        System.out.println("e. Classificação dos candidatos:");
        System.out.println("   Candidato 1: " + votosCandidato1 + " votos");
        System.out.println("   Candidato 2: " + votosCandidato2 + " votos");
        System.out.println("   Candidato 3: " + votosCandidato3 + " votos");

        System.out.println("f. Total de votos: " + totalVotos);

        scanner.close();
    }
}